feat: Added diagram of the recommended architecture to the documentation (#78)
surfstudio/surf-flutter-app-template: docs/architecture_and_project_structure.md

# Architecture

![architecture scheme](/docs/images/architecture.png)

### Presentation

> Logic in a presentation layer can be either business logic(Model) or UI logic(WidgetModel):

Reviewer: ```suggestion
Logic in a presentation layer can be either **Business logic**(Elementary Model) or **UI logic**(Widget Model):
```

> * Business logic is the implementation of product requirements for application data. For example, bookmarking an article in a news reader app when a user clicks on a button.

Reviewer: ```suggestion
* **Business logic** is the implementation of product requirements for application data. For example, bookmarking an article in a news reader app when a user clicks on a button.
```

> * UI logic is related to how to display UI state on the screen. For example, obtaining the right search bar hint when the user has selected a category, scrolling to a particular item in a list, or the navigation logic to a particular screen when the user clicks a button.

Reviewer: I don't think we can take wording right from Android Developers website.
Change the examples used here, please.

> * UI logic is related to how to display UI state on the screen. For example, obtaining the right search bar hint when the user has selected a category, scrolling to a particular item in a list, or the navigation logic to a particular screen when the user clicks a button.

Reviewer: ```suggestion
* **UI logic** is related to how to display UI state on the screen. For example, obtaining the right search bar hint when the user has selected a category, scrolling to a particular item in a list, or the navigation logic to a particular screen when the user clicks a button.
```

### Repository

> The repository is designed to abstract the details of data storage. It makes the ElementaryModel independent of the specific data source. You need to interact with the repository through the interface from the ElementaryModel.

Reviewer: ```suggestion
The repository is designed to abstract the details of data storage. It makes the `ElementaryModel` independent of the specific data source. You need to interact with the repository from the `ElementaryModel` through the interface.
```

### Presentation

Logic in a presentation layer can be either **Business logic**(Elementary Model) or **UI logic**(Widget Model):
* **Business logic** is the implementation of product requirements for application data. For example, adding an item to the cart when the user clicks on a button.

> * **UI logic** is related to how to display UI state on the screen. For example, displaying or hiding the load indicator depending on the data load status or showing a snack bar.

Reviewer: displaying or hiding load indicator is not a UI logic here. You can leave the example with snack bar and navigation.
